vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt

```diff
@@ -248,6 +293,34 @@ class RoomSettingsViewModel @AssistedInject constructor(@Assisted initialState:
         }
     }
 
+    /**
+     * Several changes are required:
+     * The new members can access only on invite
+     * The encryption has to be enabled by default
+     * The room will become private
+     * The history visibility value is replaced with invited
+     */
+    private fun handleRemoveFromRoomsDirectory() {
+        postLoading(true)
+
+        viewModelScope.launch {
+            try {
+                // Update first the joinrule to INVITE.
+                room.setJoinRuleInviteOnly()
+                // Turn on the encryption in this room (if this is not already done).
+                if (!room.isEncrypted()) room.enableEncryption()
+                // Remove the room from the room directory.
+                session.setRoomDirectoryVisibility(room.roomId, RoomDirectoryVisibility.PRIVATE)
+            } catch (failure: Throwable) {
+                postLoading(false)
+                _viewEvents.post(RoomSettingsViewEvents.Failure(failure))
+            }
+            // Update history visibility.
+            tryOrNull { room.updateHistoryReadability(RoomHistoryVisibility.INVITED) }
```

Comment:
During my tests, I observed an issue at the level of the definition of `enum class RoomHistoryVisibility `.
TchapV2 doesn't have the same definition as Element-Android

I realized this issue when I read (in the web client) this following message:
![image](https://user-images.githubusercontent.com/8969772/135168714-7c1917c3-aab2-4b47-9e16-aa9c829a4f0b.png)

Reply:
the expected message is:
![image](https://user-images.githubusercontent.com/8969772/135168816-27217b55-ab2f-40ba-a6dd-f7c2a508d9ab.png)

Reply:
here is the diff:
![image](https://user-images.githubusercontent.com/8969772/135169023-a3d8aef4-50e7-40fa-948c-052393615c72.png)

Reply:
It's an error during the rebase on element 1.1.9. I fixed it.